examples/jsm/lights/LightProbeGenerator.js

```diff
@@ -150,19 +162,28 @@ class LightProbeGenerator {
 
 			let data;
 
-			if ( dataType === HalfFloatType ) {
+			if ( useAsync ) {
 
-				data = new Uint16Array( imageWidth * imageWidth * 4 );
+				data = await renderer.readRenderTargetPixelsAsync( cubeRenderTarget, 0, 0, imageWidth, imageWidth, 0, faceIndex );
 
 			} else {
 
-				// assuming UnsignedByteType
 
-				data = new Uint8Array( imageWidth * imageWidth * 4 );
+				if ( dataType === HalfFloatType ) {
 
-			}
+					data = new Uint16Array( imageWidth * imageWidth * 4 );
 
-			renderer.readRenderTargetPixels( cubeRenderTarget, 0, 0, imageWidth, imageWidth, data, faceIndex );
+				} else {
+
+					// assuming UnsignedByteType
+
+					data = new Uint8Array( imageWidth * imageWidth * 4 );
+
+				}
+
+				renderer.readRenderTargetPixels( cubeRenderTarget, 0, 0, imageWidth, imageWidth, data, faceIndex );
```

Comment:
How about we make this call async as well?

Reply:
I'd forgotten there was an async version for WebGLRenderer. However, I have just tested it, and it seems rather slow, something that isn't replicated using the WebGL fallback oddly.

And while it is not async is there any reason for repeatedly creating the data buffer for each face?

Reply:
Also found out that the slowness with WebGLRenderer can be fixed by moving the gl.flush() to after the gl.fenceSync().

Reply:
> And while it is not async is there any reason for repeatedly creating the data buffer for each face?

That does indeed seems to be not necessary. Moving this bit out of the for loop is the better approach.